articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,563 @@
+The callable itself is then executed on a *[target machine](xref:microsoft.quantum.machines)*.
+Such target machines can be actual quantum hardware or the multiple simulators available as part of the QDK.
+For our purposes here, the most useful target machine is an instance of the [full-state simulator](xref:microsoft.quantum.machines.full-state-simulator), `QuantumSimulator`, which calculates the program's behavior as if it were being executed on a noise-free quantum computer.
+
+So far, we've described what happens when a specific Q# callable is being executed.
+Regardless of whether Q# is used in a standalone application or with a host program, this general process is more or less the same---hence the QDK's flexibility.
+The differences between the different ways of calling into the Quantum Development Kit therefore reveal themselves in *how* that Q# callable is called to be executed, and in what manner any results are returned.
+More specifically, the differences revolve around 
+1. indicating which Q# callable is to be executed,
+2. how potential callable arguments are provided,
+3. specifying the target machine on which to execute it, and
+4. how any results are returned.
+
+First, we discuss how this is done with the Q# standalone application from the command line, and then proceed to using Python and C# host programs.
+We reserve the standalone application of Q# Jupyter Notebooks for last, because unlike the first three, it's primary functionality does not center around a local Q# file.
+
+> [!NOTE]
+> Although we don't illustrate it in these examples, one commonality between the execution methods is that any messages printed from inside the Q# program (by way of [`Message`](xref:microsoft.quantum.intrinsic.message) or [`DumpMachine`](xref:microsoft.quantum.diagnostics.dumpmachine), for example) will typically always be printed to the respective console.
+
+## Q# from the command line
+One of the easiest ways to get started writing Q# programs is to avoid worrying about separate files and a second language altogether.
+Using Visual Studio Code or Visual Studio with the QDK extension allows for a seamless work flow in which we run Q# callables from only a single Q# file.
+
+For this, we will ultimately invoke the program's execution by entering
+```bash
+dotnet run
+```
+in the command line.
+The simplest workflow is when the terminal's directory location is the same as the Q# file, which can be easily handled alongside Q# file editing by using the integrated terminal in VS Code, for example.
+However, the [`dotnet run` command](https://docs.microsoft.com/dotnet/core/tools/dotnet-run) accepts numerous options, and the program can also be run from a different location by simply providing `--project <PATH>` with the location of the Q# file.
+
+
+### Add entry point to Q# file
+Most Q# files will contain more than one callable, so naturally we need to let the compiler know *which* callable to execute when we provide the `dotnet run` command.
```

Comment:
General markdownlint rules used on this repo suggest leaving a blank line after headers.
```suggestion

Most Q# files will contain more than one callable, so naturally we need to let the compiler know *which* callable to execute when we provide the `dotnet run` command.
```